h5loader

Plasmacaster opened the pull request:
Module for reading new H5 OMEGA Dante data

Plasmacaster committed:
h5loader

Module for reading new H5 OMEGA Dante data

Plasmacaster committed:
no array no NaNs

Added NaN handling for blank entries in response file. Changed docstring to reflect numpy array not compatible with function.

Plasmacaster committed:
fixed detArr bug

detector array function throws exception if boundary is y0. Added another if statement to fix this.

Plasmacaster committed:
integer reindexer

Added a function to reindex DataFrames with integers to provide compatibility with other functions. Added timesFrame docstring.

Plasmacaster committed:
PCHIP_module

Added separate module for PCHIP-based unfolds. Removed PCHIP functions from cspline. Added docstrings for PCHIP functions

lemmatum commented:
Main branch of lanl/fiducia has been updated to include new requirements (numba and h5py). Also had to update the github actions files since python version 3.7 and 3.8 are EOL.

You should be able to do a quick rebase and it'll get the tests running on this PR. Currently I'm expecting at least 3 of the pytests to fail, possibly due to the newly added numba dependency. Looks like the linter might also fail.

lemmatum commented on fiducia/h5loader.py:
```diff
@@ -0,0 +1,565 @@
+        The H5 file heirarchy and datasets.
+
+    """
+    dataImp= h5py.File(filepath)
```
It would be safer to refactor this code to use a `with` context manager.
In the current state the file might get closed improperly and could become corrupted.

lemmatum replied:
As a matter of fact, I don't see a close() being invoked anywhere...

lemmatum replied:
I've added a suggestion towards the end for how to refactor.

Plasmacaster replied:
All h5 I/O is done within the <with> context manager

lemmatum commented on fiducia/h5loader.py:
```diff
@@ -0,0 +1,565 @@
+    attrsFrame : DataFrame
+        Pandas array containing all of the attributes (scope settings) for each
+        detector position.
+    dataFrame : TYPE
```
I think the type here is `pandas.DataFrame`

lemmatum commented on fiducia/h5loader.py:
```diff
@@ -0,0 +1,565 @@
+        # get all attributes of first position
+        attrs = np.array(list(attSearch.attrs.keys()))
+    except:
+        print("Data keys and attributes location may have changed! Check file"+
```
How do you want to handle this failure? Should the program stop here or can it continue even with this failure?

If it needs to fail then use `raise Exception` instead of `print`.
If the program can continue, then this would be a good case for using the [warnings module](https://docs.python.org/3/library/warnings.html#module-warnings)

Plasmacaster replied:
I want it to stop here. If there is a file format change we can cross that bridge when we get there. I just wanted something to detect it first.

lemmatum commented on fiducia/h5loader.py:
```diff
@@ -0,0 +1,565 @@
+            print("yMult value not found.")
+    
+    # Regular expression to find the value after 'YOFF'
+    match = re.search(r'YOFF\s([-+]?\d*\.\d+([eE][-+]?\d+)?)', preamble)
```
Can this be replaced with something like `str.startswith()` or is the regex necessary? Regexes are generally more difficult to maintain, but they do have their uses.

lemmatum replied:
Same question for the following regexes.

Plasmacaster replied:
I think regex is a bit more robust. Nick uses it in his Matlab code for Dante h5, and it was recommended by chatGPT and other forums.

lemmatum commented on fiducia/h5loader.py:
```diff
@@ -0,0 +1,565 @@
+
+    """
+    # import file
+    h5File = h5_import(filepath)
```
Since `h5File` is only getting passed to `make_frames()` I think you can refactor like this:

- `make_frames()` should be the first function that gets called in `h5_rawProcess()`. It should take `filepath` as an argument instead of `h5File`.
- Handle opening the HDF5 file using the `with` context manager within the `make_frames()` function.
- Move line 462 calling `h5_import()` to be the first line under the `with` context in the `make_frames()` function.
- Change `h5_import()` so that it takes an HDF5 handle as input argument in lieu of filepath.

Plasmacaster replied:
Done!

lemmatum reviewed (changes requested):
Looks pretty good overall. Biggest required change is to add a context manager so that we don't accidentally corrupt any HDF5 files.

Plasmacaster committed:
numpy_array_handling

Added handling for channels input for responses functions to be either list or numpy array.

Plasmacaster committed:
Update .gitignore

added cpython file of new module to ignore

Plasmacaster committed:
Merge pull request #3 from Plasmacaster/Dante_upgrade_dev

Merge commits from development branch to main. This merges the new pchipSpline module as well as some bug fixes and other minor issues I picked up along the way.

Plasmacaster committed:
with statements for h5loader

Uses with statements while reading hdf5 files to avoid file corruption from not closing.

lemmatum commented on fiducia/__pycache__/pchipSpline.cpython-311.pyc:
Looks like a binary file slipped past your .gitignore
Please remove it.

Plasmacaster replied:
It did not slip past, but there is a bug somewhere using Github Desktop that is allowing it in. Check the gitignore file I SWEAR it's in there.

lemmatum reviewed (changes requested):
Remove the .pyc binary file so that we don't pollute the git history (it blows up in size trying to track binaries). After that I'll merge it.

Plasmacaster committed:
gitignore shenanignas

trying to git (see what I did there?) rid of this damn binary

lemmatum reviewed (approved):
Looks good!
